R/guides-.R

```diff
@@ -502,43 +563,47 @@ Guides <- ggproto(
 
     Map(
       grobs    = grobs,
-      position = names(grobs),
+      position = positions,
+      inside_position = inside_positions,
+      inside_just = inside_justs,
       self$package_box,
       MoreArgs = list(theme = theme)
     )
   },
 
   # Render the guides into grobs
-  draw = function(self, theme,
-                  default_position = "right",
-                  direction = NULL,
+  draw = function(self, theme, positions = NULL, direction = NULL,
                   params = self$params,
                   guides = self$guides) {
-    positions <- vapply(
+    positions <- positions %||% vapply(
       params,
-      function(p) p$position[1] %||% default_position,
-      character(1)
+      function(p) p$position[1] %||% "right",
+      character(1), USE.NAMES = FALSE
     )
-    positions <- factor(positions, levels = c(.trbl, "inside"))
-
-    directions <- rep(direction %||% "vertical", length(positions))
     if (is.null(direction)) {
-      directions[positions %in% c("top", "bottom")] <- "horizontal"
+      directions <- ifelse(
+        positions %in% c("top", "bottom"), 
+        "horizontal", "vertical"
+      )
+    } else {
+      directions <- rep(direction, length(positions))
     }
 
     grobs <- vector("list", length(guides))
     for (i in seq_along(grobs)) {
       grobs[[i]] <- guides[[i]]$draw(
-        theme = theme, position = as.character(positions[i]),
+        theme = theme, position = positions[i],
         direction = directions[i], params = params[[i]]
       )
     }
-    keep <- !vapply(grobs, is.zero, logical(1))
-    split(grobs[keep], positions[keep])
+    grobs
   },
 
-  package_box = function(grobs, position, theme) {
-
+  # here, we put `inside_position` and `inside_just` in the last, so that it
+  # won't break current implement of patchwork, which depends on the top three
+  # arguments to collect guides
+  package_box = function(grobs, position, theme,
+                         inside_position = NULL, inside_just = NULL) {
```

Comment:
I still think we could prevent having to edit this method.
We can just make a `theme` for every iteration that has the correct `inside_position` and `inside_just` baked in.

Reply:
fixed! Thanks for your suggestions!